analyzer/src/funTest/kotlin/NpmTest.kt

```diff
@@ -94,28 +92,31 @@ class NpmTest : WordSpec() {
                 yamlMapper.writeValueAsString(result) shouldBe expectedResult
             }
 
-            "abort if no lockfile is present" {
+            "show error if no lockfile is present" {
                 val workingDir = File(projectDir, "no-lockfile")
                 val packageFile = File(workingDir, "package.json")
 
-                val exception = shouldThrow<IllegalArgumentException> {
-                    NPM.create().resolveDependencies(projectDir, listOf(packageFile))
-                }
+                val result = NPM.create().resolveDependencies(projectDir, listOf(packageFile))[packageFile]
 
-                @Suppress("UnsafeCallOnNullableType")
-                exception.message!! should startWith("No lockfile found in")
+                result shouldNotBe null
+                result!!.project shouldBe Project.createEmpty()
+                result.packages.size shouldBe 0
+                result.errors.size shouldBe 1
+                result.errors.first() shouldStartWith "IllegalArgumentException: No lockfile found in"
             }
 
-            "abort if multiple lockfiles are present" {
+            "show error if multiple lockfiles are present" {
                 val workingDir = File(projectDir, "multiple-lockfiles")
                 val packageFile = File(workingDir, "package.json")
 
-                val exception = shouldThrow<IllegalArgumentException> {
-                    NPM.create().resolveDependencies(projectDir, listOf(packageFile))
-                }
+                val result = NPM.create().resolveDependencies(projectDir, listOf(packageFile))[packageFile]
 
-                @Suppress("UnsafeCallOnNullableType")
-                exception.message!! should endWith("contains multiple lockfiles. It is ambiguous which one to use.")
+                result shouldNotBe null
+                result!!.project shouldBe Project.createEmpty()
+                result.packages.size shouldBe 0
+                result.errors.size shouldBe 1
+                result.errors.first() shouldStartWith "IllegalArgumentException:"
```

Comment:
Can't we just use `should startWith("prefix")` / `should endWith("suffix")` (see [here](https://github.com/kotlintest/kotlintest/blob/master/doc/matchers.md))?

Reply:
Done, had to switch to FreeSpec for this.